Tests/SwiftFormatRulesTests/NoAccessLevelOnExtensionDeclarationTests.swift

```diff
@@ -49,6 +49,30 @@ public class NoAccessLevelOnExtensionDeclarationTests: DiagnosingTestCase {
     )
   }
 
+  public func testPreservesCommentOnRemovedModifier() {
+    XCTAssertFormatting(
+      NoAccessLevelOnExtensionDeclaration.self,
+      input: """
+        /// This doc comment should stick around.
+        public extension Foo {
+        }
+
+        /// So should this one.
+        internal extension Foo {
```

Comment:
It might be helpful to add a test case with members to show that the comment isn't traveling with the keyword.